README.md

```diff
@@ -0,0 +1,73 @@
+
+### Targets
+
+Each website is called a target. All targets are configured in `config.yml`. Each websites display name is the key of the `targets` dictionary.   
+Each target must at least contain the following keys:
+
+- `href`: Single value containing a link to the websites front page
+- `hosts`: A list containing at least one domain to check for AAAA records. Only include domains where website ressources are gathered from. Skip CND domains.
+
+The following keys are optional:
+
+- `icon`: Link to an FontAwesome glyph or a picture. See section `Adding icons` for more details
+- `twitter`: If the website has one or more twitter handles you can add them here. Don't forget to add quotes! Multiple twitter handles are seperated by a whitespace.
+- `categories`: Contains a list with category names. This is used to render a host into the respective categories. A target can be part of multiple categories. If this key is missing the target will be automatically grouped into the category `Uncategorized`
+
+### Adding icons
+
+Adding icons helps to quickly identify a target.
+
+#### By using FontAwesome
+
+If you find an icon provided by [FontAwesome](https://fontawesome.com/icons?d=gallery&s=brands) you can add it by setting the `icon` key of a target to `fa-iconname`. The page renderer will recognize the `fa`-Prefix and will create a glyph.
+
+#### By providing an image
+
+If you can provide an image for a target you can add it to `dist/images/`. Afterwards set the `icon` key to the filename.   
+Some websites provide their official logos and their guidelins of how to use them. You can find them by using your favorite search engine. Some search terms to find them:
+
+- `website logo guidelines`
+- `website logo press center`
+- `website logo press media`
+
+If you find no good source try to contact the website owner (for example on Twitter) and ask them if they have guidelines and if you are allowed to use their logo.   
+Please add the image source near the `icon` key and (if available) a link to the licence, guideline and/or permission. Please don't manipulate images in any way if it's forbidden.
+
+Before you add the file to the repository make sure you optimize the file size to save bandwidth and speed up the page. For PNG files have a look at `optipng`. For JPG/JPEG files have a look at `jpegoptim`. Most images can be reduced by more than 80% in file size.
+
+### Formatting config.yml
+
+YAML files are sometimes hard to format properly. You can use `yamlfmt` to format `config.yml` it properly before checking it in: `yamlfmt -w config.yml`
+
+### Minifying HTML
+
+When you generate the static page it's very huge. It consumes round about 1.3 MByte disk space and consists of many blank spaces which are good for a human developer working and debugging the generated HTML. A computer doesn't need them and only wastes ressources on it. That's why minifiers were developed. They remove unnecessary spaces, newlines etc to save up space.   
```

Comment:
Did you check how much influence compression has on that? I suspect it might be very effective.

Reply:
Yeah it's huge. I just checked: All Images (after optimizing most of the png and jpg files) are 1.5 MByte in total. Add the 1.3 MByte from the unminified page you have 2.8 MByte. With my internet connection no problem, but speeds lower than 25 MBit is over a second if the downloads would start at full speed. CSS, JS and webfonts not included.

Also the rendering takes less CPU which is good for mobile phones and tablets.